addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -97,108 +98,663 @@ def general_ledger_report(self, account_id, start_date, end_date):
             'ledger_data': ledger_data,
         }
 
+    ##create/get/delete_bills
+    @api.model
+    def create_bill(self, invoice_vals, partner_id, invoice_date, invoice_date_due, reference, narration):
+        """
+        Create an AR invoice and corresponding analytic lines based on the provided data.
+        """
+        # Prepare the invoice lines
+        invoice_lines = []
+        for line in invoice_vals:
+            invoice_line_vals = {
+                'name': line.get('description', ''),
+                'quantity': line.get('quantity', 1.0),
+                'price_unit': line.get('price_unit', 0.0),
+                'account_id': line.get('account_id'),
+                'product_id': line.get('product_id', False),
+            }
+            invoice_lines.append((0, 0, invoice_line_vals))
+
+        # Create a new AR invoice record
+        invoice = self.env['account.move'].create({
+            'move_type': 'in_invoice',
+            'partner_id': partner_id,
+            'invoice_date': invoice_date,
+            'invoice_date_due': invoice_date_due,
+            'ref': self.name,
+            'narration': narration,
+            'invoice_line_ids': invoice_lines,
+        })
+        invoice.action_post()
+
+        # Create analytic lines for each invoice line if analytic_account_id is provided
+        for line, line_vals in zip(invoice.invoice_line_ids, invoice_vals):
+            analytic_account_id = line_vals.get('analytic_account_id')
+            if analytic_account_id:
+                self.env['account.analytic.line'].create({
+                    'account_id': analytic_account_id,
+                    'name': line.name,
+                    'amount': line.price_subtotal,  # or any other relevant amount
+                    # 'move_id': line.id,
+                    # Add other necessary fields
+                })
+
+        return f"ID: {invoice.id}, Name: {invoice.name}, Amount: {invoice.amount_total}, Date: {invoice.invoice_date}, Due Date: {invoice.invoice_date_due}"
```

Comment:
@sameerz09 data should be returned as structured json. Why are we not doing that?